### ISSUE TITLE
nixos.tests.networking.scripted.macvlan.x86_64-linux fails, blocking the channels

### ISSUE THREAD

**Issue by doronbehar** [NixOS/nixpkgs#102482]
**Describe the bug**

https://hydra.nixos.org/job/nixos/trunk-combined/nixos.tests.networking.scripted.macvlan.x86_64-linux fails.

**To Reproduce**

```sh
nix build -f. -L nixosTests.networking.scripted.macvlan
```

It will not fail immediately, but it will eventually, and it should not take long at all (see e.g for a [successful test](https://nix-cache.s3.amazonaws.com/log/m4si4pbh2kq2g5m7qndxyxh2br2nk7l4-vm-test-run-MACVLAN-Networking-Scripted.drv)).

**Additional context**

Couldn't find an interesting commit in the git log since the last successful test:

https://hydra.nixos.org/api/scmdiff?rev2=4e57249fc365bd8b0e319e1dbcf47f5efb8b9395&uri=https%3A%2F%2Fgithub.com%2FNixOS%2Fnixpkgs.git&branch=&rev1=800ebd4561d6d4470fc53b51766fb1a600b65488&type=git

**Notify maintainers**

@flokli ?

**Comment by vcunat** [comment]
Works for me locally.  I restarted the job.

**Comment by vcunat** [comment]
BTW, the big channels are now mainly blocked by https://github.com/NixOS/nixos-org-configurations/issues/128

**Comment by doronbehar** [comment]
Hmmm... Didn't work for me in the first attempt among 3.

**Comment by vcunat** [comment]
Maybe it's still worth looking into reliability of the test 🤷🏽   but occasional restarts aren't such a huge burden, someone has to watch for breakages anyway.

**Comment by erosennin** [comment]
The test is blocking nixos-unstable again.

https://hydra.nixos.org/build/136640438

**Comment by vcunat** [comment]
It was just a single attempt on Hydra and the second one succeeded.  Locally it succeeded 1/1 again; perhaps it's more prone to fail on a loaded system.  Well, my last comment still holds.

**Comment by erosennin** [comment]
When I run the test locally, it fails (or rather gets stuck) once in five tries or so:

```
while true; do nix -vL build --rebuild -f. nixosTests.networking.scripted.macvlan; done
```

It will fail eventually, as @doronbehar said.

**Comment by erosennin** [comment]
Seems like this is caused by two different interfaces getting the same IP via DHCP.

From the [failed test log](https://hydra.nixos.org/build/135856814/nixlog/1):

```
client: must succeed: ip addr >&2
client # 1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
client #     link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
client #     inet 127.0.0.1/8 scope host lo
client #        valid_lft forever preferred_lft forever
client #     inet6 ::1/128 scope host
client #        valid_lft forever preferred_lft forever
client # 2: eth0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN group default qlen 1000
client #     link/ether 52:54:00:12:34:56 brd ff:ff:ff:ff:ff:ff
client # 3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
client #     link/ether 52:54:00:12:01:01 brd ff:ff:ff:ff:ff:ff
client #     inet 192.168.1.2/24 brd 192.168.1.255 scope global dynamic noprefixroute eth1
client #        valid_lft 599sec preferred_lft 524sec
client #     inet6 fd00:1234:5678:1::2/128 scope global dynamic noprefixroute
client #        valid_lft 596sec preferred_lft 371sec
client #     inet6 fe80::5054:ff:fe12:101/64 scope link
client #        valid_lft forever preferred_lft forever
client # 4: macvlan@eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
client #     link/ether 76:d5:5f:3b:0b:1d brd ff:ff:ff:ff:ff:ff
client #     inet 192.168.1.2/24 brd 192.168.1.255 scope global dynamic noprefixroute macvlan
client #        valid_lft 596sec preferred_lft 521sec
client #     inet6 fe80::74d5:5fff:fe3b:b1d/64 scope link
client #        valid_lft forever preferred_lft forever
```

Both `eth1` and `macvlan@eth1` have IP `192.168.1.2`, so `ping 192.168.1.3` fails.

[Successful test log](https://hydra.nixos.org/build/136779925/log) looks like this:

```
client: must succeed: ip addr >&2
client # 1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
client #     link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
client #     inet 127.0.0.1/8 scope host lo
client #        valid_lft forever preferred_lft forever
client #     inet6 ::1/128 scope host
client #        valid_lft forever preferred_lft forever
client # 2: eth0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN group default qlen 1000
client #     link/ether 52:54:00:12:34:56 brd ff:ff:ff:ff:ff:ff
client # 3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
client #     link/ether 52:54:00:12:01:01 brd ff:ff:ff:ff:ff:ff
client #     inet 192.168.1.2/24 brd 192.168.1.255 scope global dynamic noprefixroute eth1
client #        valid_lft 593sec preferred_lft 518sec
client #     inet6 fe80::5054:ff:fe12:101/64 scope link
client #        valid_lft forever preferred_lft forever
client # 4: macvlan@eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
client #     link/ether 62:b5:89:12:9c:88 brd ff:ff:ff:ff:ff:ff
client #     inet 192.168.1.3/24 brd 192.168.1.255 scope global dynamic noprefixroute macvlan
client #        valid_lft 599sec preferred_lft 524sec
client #     inet6 fd00:1234:5678:1::2/128 scope global dynamic noprefixroute
client #        valid_lft 595sec preferred_lft 370sec
client #     inet6 fe80::60b5:89ff:fe12:9c88/64 scope link
client #        valid_lft forever preferred_lft forever
```

Here, `macvlan@eth1` has IP `192.168.1.3` as it is supposed to.

**Comment by erosennin** [comment]
It failed again: https://hydra.nixos.org/build/136936565.

Could somebody please look at  #112674?

**Comment by grahamc** [comment]
We should probably either make the test reliable or delete the test.

**Comment by grahamc** [comment]
It passed. We should probably open a new issue if it fails again, with a focus on trying to stabilize the test .

**Comment by erosennin** [comment]
@grahamc I believe #112674 fixes it. Not sure why nobody wants to merge it. Am I shadowbanned? :)
